### ISSUE TITLE
[V2] Merge options broken when using a top bar button with a custom component

### ISSUE THREAD

**Issue by steve228uk** [wix/react-native-navigation#4094]
### Issue Description

When using the `Navigation.mergeOptions` method to add a button to the `topBar` with a custom component an error is thrown.

### Steps to Reproduce / Code Snippets / Screenshots

I'm currently using:

```js
 Navigation.mergeOptions({
    topBar: {
        rightButtons: [
            {
                id: 'leaderboardButton',
                component: {
                    name: 'explore.leaderboard-button',
                    passProps: {
                        onPress: () => { this.push('explore.leaderboard') }
                    }
                }
            }
        ]
    }
});
```

Which throws an error from the `[RNNTabBarController Creator]`.

---
### Environment
* React Native Navigation version: 2.0.2573
* React Native version: 0.56.0
* Platform(s) (iOS, Android, or both?): iOS
* Device info (Simulator/Device? OS version? Debug/Release?): All

**Comment by steve228uk** [comment]
Reverting to `2.0.2569` fixed this issue.

**Comment by steve228uk** [comment]
This seems to be resolved in `2.0.2577`
